src/datasets/utils/py_utils.py

```diff
@@ -439,11 +440,29 @@ def map_nested(
 
     if num_proc is None:
         num_proc = 1
-    if num_proc <= 1 or len(iterable) < parallel_min_length:
+
+    if (num_proc <= 1 and not use_spark) or len(iterable) < parallel_min_length:
         mapped = [
             _single_map_nested((function, obj, types, None, True, None))
             for obj in logging.tqdm(iterable, disable=disable_tqdm, desc=desc)
         ]
+    elif use_spark:
+        try:
+            import pyspark
+
+            spark = pyspark.sql.SparkSession.builder.getOrCreate()
+            mapped = (
+                spark.sparkContext.parallelize(iterable)
+                .map(lambda obj: _single_map_nested((function, obj, types, None, True, None)))
+                .collect()
+            )
+
+        except ImportError:
```

Comment:
We may need to provide some more details for the exceptions. If the import failed, we need to give a warning saying something like "pyspark module is not found. Downloading in sequentially". If spark session creation fails, we need to give corresponding messages.